### PR TITLE
feat(admin): Enable/Disable Recaptcha Option #2376

### LINKED ISSUE #2376: Enable/Disable Recaptcha Option
## Issue Overview
While working on issue #2304 and #2023, I've noticed that enabling and disabling the recaptcha is not possible once the `Site Key` and `Secret Key` os Recaptcha is added. It is only possible if user removes the keys from the system.

image

## Expected Behavior
There should be option to enable/disable Recaptcha with the key already present in the database. This process will help in faster switching and quick debugging.

## Current Behavior
Currently, Recaptcha is dependent on Email Access toggle which is not good alternative.

## Todos
- [ ] Fix
- [ ] Tests
- [ ] Documentation

## WordPress Environment
<details>

### WordPress Environment ###

Home URL: http://givebase.local
Site URL: http://givebase.local
WP Version: 4.9
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: ✔
WP Cron: ✔
Language: en_US
Permalink Structure: /%year%/%monthnum%/%day%/%postname%/
Show on Front: posts
Table Prefix Length: 3
Table Prefix Status: Acceptable
Admin AJAX: Inaccessible
Registered Post Statuses: publish, future, draft, pending, private, trash, auto-draft, inherit, refunded, failed, revoked, cancelled, abandoned, processing, preapproval, give_subscription

### Server Environment ###

Hosting Provider: DBH: localhost, SRV: givebase.local
TLS Connection: Connection uses TLS 1.2
TLS Connection: Probably Okay
Server Info: nginx/1.10.1
PHP Version: ❌ 5.3.29 - We recommend a minimum PHP version of 5.6. See: PHP Requirements in Give
PHP Post Max Size: 1,000 MB
PHP Time Limit: 900
PHP Max Input Vars: 3000
PHP Max Upload Size: 1,000 MB
cURL Version: ❌ 7.38.0, OpenSSL/1.0.2h - We recommend a minimum cURL version of 7.40.
SUHOSIN Installed: –
MySQL Version: 5.6.34
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
gzip: ✔
GD Graphics Library: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Give Configuration ###

Give Version: 1.8.17
Upgraded From: 1.8.17
Test Mode: Disabled
Currency Code: USD
Currency Position: Before
Decimal Separator: .
Thousands Separator: ,
Success Page: http://givebase.local/donation-confirmation/
Failure Page: http://givebase.local/donation-failed/
Donation History Page: http://givebase.local/donation-history/
Give Forms Slug: /donations/
Enabled Payment Gateways: Test Donation, Offline Donation
Default Payment Gateway: Test Donation
PayPal IPN Verification: Enabled
PayPal IPN Notifications: N/A
Admin Email Notifications: Disabled
Donor Email Access: Enabled

### Session Configuration ###

Give Use Sessions: Enabled
Session: Disabled

### Active Give Add-ons ###

Give - Form Field Manager: ❌ Unlicensed – by WordImpress – 1.2.4
Give - Manual Donations: ❌ Unlicensed – by WordImpress – 1.2
Give - Recurring Donations: ❌ Unlicensed – by WordImpress – 1.4

### Other Active Plugins ###

Duplicate Post: by Enrico Battocchi – 3.2
Mehul Customization: by  – 1.0

### Inactive Plugins ###

WP Default Notices: by  – 1.0.0

### Theme ###

Name: Twenty Seventeen
Version: 1.3
Author URL: https://wordpress.org/
Child Theme: No – If you're modifying Give on a parent theme you didn't build personally, then we recommend using a child theme. See: How to Create a Child Theme

</details>

### PR DESCRIPTION
**Pull request by Sidsector9**
Closes #2376 

## Description
Added a checkbox to **enable/disable** reCAPTCHA without the need to delete the _Site Key_ and the _Secret Key_

## How Has This Been Tested?
This has been manually tested

## Screenshots
recaptcha

## Types of changes
New feature:
Added a checkbox at - _**Donations > Settings > General > Access Control > Enable reCAPTCHA**_

## Checklist:
- [ ] My code is tested.
- [ ] My code follows the WordPress code style.
- [ ] My code follows has proper inline documentation.

### CODE REVIEW

**Comment by Sidsector9** [comment]
As discussed with @ravinderk over the call, the previous commit has been reverted with a new one without running Gulp

**Comment by raftaar1191** [comment]
@Sidsector9 

Change checkbox to radio button as we are using it for all our setting 

image

File: includes/login-register.php [hunk 1]
```diff
@@ -325,7 +325,9 @@ function give_email_access_login() {
 
 	$recaptcha_key    = give_get_option( 'recaptcha_key' );
 	$recaptcha_secret = give_get_option( 'recaptcha_secret' );
-	$enable_recaptcha = ! empty( $recaptcha_key ) && ! empty( $recaptcha_secret ) ? true : false;
+	$enable_recaptcha = give_get_option( 'enable_recaptcha' );
+
+	$enable_recaptcha = ( 'on' === $enable_recaptcha ) && ! empty( $recaptcha_key ) && ! empty( $recaptcha_secret ) ? true : false;
```
Comment (raftaar1191) [inline includes/login-register.php]: Kindly used `give_is_setting_enabled` function to check if reCAPTCHA is Enable or not

File: templates/email-login-form.php [hunk 2]
```diff
@@ -15,7 +15,9 @@
 
 $recaptcha_key    = give_get_option( 'recaptcha_key' );
 $recaptcha_secret = give_get_option( 'recaptcha_secret' );
-$enable_recaptcha = ! empty( $recaptcha_key ) && ! empty( $recaptcha_secret ) ? true : false;
+$enable_recaptcha = give_get_option( 'enable_recaptcha' );
+
+$enable_recaptcha = ( 'on' === $enable_recaptcha ) && ! empty( $recaptcha_key ) && ! empty( $recaptcha_secret ) ? true : false;
```
Comment (raftaar1191) [inline templates/email-login-form.php]: Kindly used `give_is_setting_enabled` function to check if reCAPTCHA is Enable or not

**Comment by Sidsector9** [comment]
@raftaar1191 Updated

**Comment by raftaar1191** [comment]
@ravinderk will you like if we hide the reCAPTCHA site key and Secret Key if the **Enable reCAPTCHA** is **Disabled**


image

**Comment by raftaar1191** [comment]
@Sidsector9 
Nice PR is working Awesome Now

**Comment by ravinderk** [comment]
> Slack Chat Summary
> 
> Participants: @Sidsector9 , @ravinderk
> Topic: Toggle recaptcha setting fields on recaptcha toggle
> Result: @Sidsector9 will edit js to achieve it.

File: assets/js/admin/admin-scripts.js [hunk 3]
```diff
@@ -618,6 +618,23 @@ var give_setting_edit = false;
 			var emailAccess = $( 'input[name="email_access"]', '.give-setting-tab-body-general' );
 			emailAccess.on( 'change', function() {
 				var fieldValue = $( 'input[name="email_access"]:checked', '.give-setting-tab-body-general' ).val();
+				if ( 'enabled' === fieldValue ) {
+					$( '#recaptcha_key' ).parents( 'tr' ).show();
```
Comment (raftaar1191) [inline assets/js/admin/admin-scripts.js]: @Sidsector9 
We should used `give-hidden` to show or hide the fields

**Pull request merged**